packages/hash/frontend/src/components/hooks/blockProtocolFunctions/knowledge/useBlockProtocolUpdateEntity.ts

```diff
@@ -0,0 +1,114 @@
+import { useLazyQuery, useMutation } from "@apollo/client";
+import { useCallback } from "react";
+
+import {
+  GetEntityTypeRootedSubgraphQuery,
+  GetEntityTypeRootedSubgraphQueryVariables,
+  UpdatePersistedEntityMutation,
+  UpdatePersistedEntityMutationVariables,
+} from "../../../../graphql/apiTypes.gen";
+import { getEntityTypeRootedSubgraphQuery } from "../../../../graphql/queries/ontology/entity-type.queries";
+import { updatePersistedEntityMutation } from "../../../../graphql/queries/knowledge/entity.queries";
+import { UpdateEntityMessageCallback } from "./knowledge-shim";
+
+export const useBlockProtocolUpdateEntity = (
+  readonly?: boolean,
+): {
+  updateEntity: UpdateEntityMessageCallback;
+} => {
+  const [updateFn] = useMutation<
+    UpdatePersistedEntityMutation,
+    UpdatePersistedEntityMutationVariables
+  >(updatePersistedEntityMutation);
+
+  const [getEntityTypeRootedSubgraphFn] = useLazyQuery<
+    GetEntityTypeRootedSubgraphQuery,
+    GetEntityTypeRootedSubgraphQueryVariables
+  >(getEntityTypeRootedSubgraphQuery);
+
+  const updatePersistedEntity: UpdateEntityMessageCallback = useCallback(
+    async ({ data }) => {
+      if (readonly) {
+        return {
+          errors: [
+            {
+              code: "FORBIDDEN",
+              message: "Operation can't be carried out in readonly mode",
+            },
+          ],
+        };
+      }
+
+      if (!data) {
+        return {
+          errors: [
+            {
+              code: "INVALID_INPUT",
+              message: "'data' must be provided for updatePersistedEntity",
+            },
+          ],
+        };
+      }
+
+      const { entityId, updatedProperties } = data;
+      const { data: responseData } = await updateFn({
+        variables: {
+          entityId,
+          updatedProperties,
+        },
+      });
+
+      const { updatePersistedEntity: updatedEntity } = responseData ?? {
+        updatePersistedEntity: null,
+      };
```

Comment:
Can't we just do this instead?
```suggestion
      const { updatePersistedEntity: updatedEntity } = responseData ?? {};
```